### DISCUSSION TITLE
Include relation in entity generated type

### DISCUSSION THREAD

**Discussion by zecka** [prisma/prisma#10989] [Q&A] [answered]
Any way to add Relation in generate type from model ?

For example i have this model :
```prisma
model Client {
  id        Int        @id @default(autoincrement())
  name      String     @unique
  settings  Json
  campaigns Campaign[]
}
```

Generated type is as follow:
```ts
export type Client = {
  id: number
  name: string
  settings: Prisma.JsonValue
}
```

Is there any way to add also relation in type, so the type will be as follow:
```ts
export type Client = {
  id: number
  name: string
  settings: Prisma.JsonValue,
  campaigns?: Campaign[];
}
```

Actually I create specific type manually
```ts
export type ClientWithCampaign = Client & {
  campaigns: Campaign[];
};
```

**Comment by austincrim** [comment, marked as answer]
Hey @zecka 👋 ,

We generate the types without relations because that's how the data comes back in queries unless you explicitly use the `include` option.

I just answered a similar question [here](https://github.com/prisma/prisma/discussions/10928), but you could define your own type like this:
```ts
import { Prisma } from '@prisma/client'

type ClientWithCampaigns = Prisma.Client<{include: { campaigns: true }}>
```

Let me know if you have questions about that!

**Reply by zecka** [reply]
Awesome 😎 thanks for your quick answer

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
